fixes #904 Implement `--info`

lahma opened the pull request:
* Adds HostEnvironmentInfo.GetInformation() to get the string presentation 
* Consolidates printing logic to summary

AndreyAkinshin commented on src/BenchmarkDotNet/Environments/HostEnvironmentInfo.cs:
```diff
@@ -109,5 +113,22 @@ public override IEnumerable<string> ToFormattedString()
+        [PublicAPI]
+        public static string GetInformation()
+        {
+            var summary = new Summary(
```
I don't like an idea to create a fake summary to print information about the environment, it looks like a hack. We have to refactor it somehow.

AndreyAkinshin commented on src/BenchmarkDotNet/Exporters/AsciiDocExporter.cs:
```diff
@@ -17,11 +17,7 @@ private AsciiDocExporter()
         public override void ExportToLog(Summary summary, ILogger logger)
         {
             logger.WriteLine("....");
-            foreach (string infoLine in summary.HostEnvironmentInfo.ToFormattedString())
```
This change will conflict with other PR: https://github.com/dotnet/BenchmarkDotNet/pull/893

lahma commented:
@AndreyAkinshin thank you for the review! I reverted the conflicting changes and removed that refactoring. I changed the string building to use Summary's logic by providing minimal information required.

Waiting for your feedback and happy to tweak more where needed.

AndreyAkinshin commented:
@lahma, now it looks much better, thanks! Could you also add information about this feature in the documentation?
@adamsitnik, could you take a look?

lahma commented:
I've added info about the new switch to console-args.md.

adamsitnik commented:
@lahma thanks a lot!

/cc @eerhardt
